src/components/fields/location-field/types.ts

```diff
@@ -17,7 +18,8 @@ export interface ILocationFieldSchema<V = undefined>
 			| "disableTextSearch"
 		>,
 		Pick<ILocationInputProps, "locationInputPlaceholder" | "disabled" | "readOnly">,
-		Pick<IStaticMapProps, "staticMapPinColor"> {
+		Pick<IStaticMapProps, "staticMapPinColor">,
+		Pick<ILocationModalProps, "disableLocationSelectionOnStart"> {
```

Comment:
as `disableLocationSelectionOnStart` is applying a number of behaviours, may I suggest to introduce `selectionMode` instead? it will accept `"default" | "pins" | "undefined"`  with `pins` being the new behaviour you are introducing

Reply:
updated